### DISCUSSION TITLE
hooking up an SWG flow switch to the MEGA-BAS?

### DISCUSSION THREAD

**Discussion by johnny2678** [tagyoureit/nodejs-poolController#425] [REM]
I'm troubleshooting constant low temp/low flow error on my SWG.  Have tried switching out the flow sensor but they keep coming back.  Pool is 80 degrees and flow is good with no air bubbles and running at 2600-2900 rpm, so should have any problems with temp/flow.

Has anyone tried hooking up the 4 wire flow sensor to the inputs on the BAS to see if you get a reading?  

this diagram makes it look possible.  Says the black/red wires are flow and the green/white wires are a 10k thermister
https://www.troublefreepool.com/wiki/index.php?title=Pentair_Intellichlor_ICXX_SWG#Replacing_Bad_Temperature_Sensor_in_the_Flow_Switch_with_a_Separate_Sensor

Was thinking I should be able to hook up the black/red wires to a dry contact input and the green/white wires to a 10K input and get readings but so far no luck.

Testing with an older flow switch that's not in use but receiving a new one today and wanted to see if I was getting a good signal on both flow/temp before installing.

Is this making sense?  Should I be getting good flow/temp readings in REM with this sensor?

**Comment by johnny2678** [comment]
Ok, this deserves some ridicule... the whole time I'm testing and not getting results I'm thinking the SWG flow switch is just dead.  Then I hook up a known good 10k thermister and a chem flow cell switch that I had working before in my "lab" and I'm getting no readings from that either.  That's when I noticed that somehow my MEGA-BAS I2C device was not set to active.  🤦‍♂️ 

So now it is set to active and I get readings for both the flow and the temp from the 4 wire SWG flow switch.  Temp readings are not , "Low Temp" - reading perfect room temp.  Flow readings work when the metal spring is pressed all the way against the flow switch body.  No way to tell how much my actual pool flow is bending the spring but it doesn't take much to bend it.  Can't imagine it's not making contact under normal flow.

Which begs the question.  If the flow switch is working fine but SWG is still reporting low flow/low temp.... ???

**Comment by rstrouse** [comment]
Unfortunately, your SWG can report "Cold Water" or "Low Flow" for conditions that are neither.  If you want a head scratcher there is a bit reserved for communication lost on comms message.  Not quite sure who they would send that to.  Maybe that inactive port you set up (sorry I couldn't help myself).

I would try to jumper that flow switch.  Not completely sure of the wiring but I would guess you would simply tie the wires that would normally go to the flow switch together.

**Reply by johnny2678** [reply]
> I would try to jumper that flow switch.

meaning, use an external flow switch and external temp sensor?  thought about that...

**Reply by rstrouse** [reply]
No just to see if there is another issue with the cell.  You can do this temporarily just to isolate the problem.  Unfortunately, when the cells get old they just stop working and report the low flow even when they have flow.  

I had a spare cell sitting around and a neighbor had his cell fry so I installed the new on in place of a 12 year old cell and gave him my old one until stock came back.  That cell is still generating for him.   Wanna bet that they just stop the generation after it has run for so many hours?  

When (if) that cell dies it is going to get torn apart to determine what if any damage there was.  I cleaned that cell with acid every few months like clockwork.  I know that the folks on TFP jump up and down that you are destroying the cell by cleaning it with acid, but I am not so sure which end they are talking out of.  MA is not one of the things that is reactive with titanium and the coatings are not reactive either so not sure where their info comes from.  Ruthenium is also immune to HCL at concentrations far greater than the MA we use.

**Comment by johnny2678** [comment]
yea, I'm not really sure where I'm at with this thing.  I closed the flow circuit but nothing changed.  Right now I'm stuck in a loop where it cycles between "Low Flow", "Comm Lost", and "Ok" every few seconds and the lights on the unit are flashing like it's just starting up.  if I pull power, it just finds this state again.  At some today point it will snap out of it and start producing but no telling what's causing that.  

Also tried pulling the com port connector on my swg power center, checked contact with wires leading back to OCP - both ends of the connection looked good.

Big $$ gamble thinking a new unit will fix this problem. So before I sink $1K into a new unit, anything else you can think of first?

**Reply by johnny2678** [reply]
crazy - I bought my current cell for $550 (Amazon) in July of 2016.

**Reply by rstrouse** [reply]
Yeah I think they have a planned obsolescence built into the electronics where it will fail after so many hours which did not exist on earlier models.  I remember seeing something where you can do the push button dance and get the cell life.

**Reply by johnny2678** [reply]
oddly enough, the unit only flashes the 20% and 40% buttons when I press the more button.  I think that only corresponds to ~4000 hours.  Just opened the cell up over the weekend and there's no scale inside.  One of the plates did look a little different - maybe some corrosion starting?  should have took a pic but didn't...

**Reply by rock-crusher** [reply]
> Yeah I think they have a planned obsolescence built into the electronics where it will fail after so many hours which did not exist on earlier models. 

Lol… yep, I totally agree. I believe many electronic devices do indeed have some sort of planned obsolescence built into the electronics, smoke & carbon monoxide detectors come to mind.
